Next review iteration (#207)
cossacklabs/product-docs: acra/security-controls/encryption/_index.md

# Application-level encryption

> Acra can be configured to prevent adversaries with access to database from seeing sensitive plaintexts.

Reviewer: ```suggestion
Acra prevents adversaries with access to database from seeing sensitive data in plaintext.
```

let's be more assertive in our docs

Acra selectively encrypts sensitive data using securely stored cryptographic keys. Thus, Acra acts as a security boundary in front of sensitive data.

> Essentially, Acra provides [application-level encryption](https://www.infoq.com/articles/ale-software-architects/) by implementing it as a field-level database encryption.

Reviewer: ```suggestion
Essentially, Acra provides [application-level encryption](https://www.infoq.com/articles/ale-software-architects/) by implementing it as a field-level encryption before data gets into the database.
```

"field-level encryption" is a search term, let's not ruin it.

## Encryption

Acra encrypts data using two special cryptographic containers at choice: AcraStructrs and AcraBlocks.
Refer to [Acra in depth / Data structures](/acra/acra-in-depth/data-structures/) to learn the differences and to select suitable for you (use AcraBlock by default).

There are three ways to encrypt data:

* AcraServer transparent encryption: if AcraServer has been set up in a transparent mode, encryption happens on AcraServer during INSERT/UPDATE queries. You configure which fields should be encrypted in `encryptor_config`.
* AcraTranslator: encryption happens as a result of gRPC or HTTP API call from client application.
* [Client-side encryption using AcraWriter SDK](/acra/acra-in-depth/architecture/sdks/acrawriter/): it is possible to generate AcraStructs/AcraBlocks using AcraWriter library inside your application using appropriate keys.

## Decryption

> There are three ways to decrypt data from AcraStructs/AcraBlocks back into a plaintext:

Reviewer: good catch!

You can also encrypt the data manually on the client side using
[AcraWriter SDK](/acra/acra-in-depth/architecture/sdks/acrawriter/).

> Decryption on the other hand is done automagically: AcraServer will detect encrypted data and will attempt

Reviewer: that was a joke 😿

> Decryption on the other hand is done automatically: AcraServer will detect encrypted data and will attempt

Reviewer: that was a joke :)
